### PR TITLE
add module attribution to dcl packages

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes https://github.com/hashicorp/terraform-provider-google/issues/8204
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: tpgtools/templates/provider_dcl_client_creation.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package google
+
+import (
+	dcl "github.com/GoogleCloudPlatform/declarative-resource-client-library/dcl"
+	{{range $index, $pkg := .}}
+	{{$pkg.PackageName}} "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/{{$pkg.DCLPackage}}"
```
Comment (slevenick) [inline tpgtools/templates/provider_dcl_client_creation.go.tmpl:33]: What about beta vs GA clients here? This will mismatch when compiled for beta

**Review by slevenick — COMMENTED** [review]
Looks like there will be a mismatch between beta vs GA dependencies, but other than that this looks great!

Seems like a clean way to handle new clients per request

File: tpgtools/templates/provider_dcl_client_creation.go.tmpl (hunk 1 above)
Comment (ScottSuarez) [reply tpgtools/templates/provider_dcl_client_creation.go.tmpl:33]: that should already be accounted for
https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4814-old..auto-pr-4814

File: tpgtools/templates/resource.go.tmpl [hunk 2]
```diff
@@ -413,7 +428,12 @@ func resource{{$.PathType}}Delete(d *schema.ResourceData, meta interface{}) erro
 {{- end}}
 
 	log.Printf("[DEBUG] Deleting {{$.Type}} %q", d.Id())
-	if err := config.client{{$.ProductType}}DCL.Delete{{$.Type}}(context.Background(), obj); err != nil {
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+	client := Create{{$.ProductType}}Client(config, userAgent, "")
```
Comment (slevenick) [inline tpgtools/templates/resource.go.tmpl:435]: What do we need to do to get the billing project here? Passing `""` seems strange, maybe remove this parameter until we are ready to use it?

**Review by slevenick — APPROVED** [review]

File: tpgtools/templates/resource.go.tmpl (hunk 2 above)
Comment (ScottSuarez) [reply tpgtools/templates/resource.go.tmpl:435]: It's another task I have in tandem to add support for billing project. It was a little throttled by figuring out if we should upstream it to the dcl or not. I can remove it

**Pull request merged**